### PR TITLE
(maint) Update references in master configuration docs

### PR DESCRIPTION
**Pull request by justinstoller**
This is a re-write and might be better to just view the doc instead of the diff.

### CODE REVIEW

**Comment by justinstoller** [comment]
@baurmatt do you think this is an improvement?
@Magisus & @mwaggett can you check this?

**Review by Magisus — DISMISSED** [review]
This is great, much clearer for the current world. All the links look right to me.

File: doc/dynamic-environments/master-configuration.mkd [hunk 1]
```diff
@@ -1,70 +1,40 @@
...
-## Puppet <= 3.4.x
+This is the default behavior in Puppet 4+
 
-In Puppet 3.4 and earlier, dynamic environments are achieved by interpolating
-the `$environment` variable in the modulepath, manifestdir, and manifest
-settings. When environments is loaded the settings are dynamically set based on
-the name of the environment, thus allowing environments to be created on the
-fly.
+This behavior is controled by the following settings
+(listed with their default values)
```
Comment (mwaggett) [inline doc/dynamic-environments/master-configuration.mkd]:
```suggestion
(listed with their default values):
```

File: doc/dynamic-environments/master-configuration.mkd [hunk 2]
```diff
@@ -1,70 +1,40 @@
...
+[Directories and the Modulepath](https://puppet.com/docs/puppet/latest/dirs_modulepath.html).
+More information can be found about environments in general at
+[Creating Environmnets](https://puppet.com/docs/puppet/latest/environments_creating.html).
 
-    [master]
-    # None of modulepath, manifestdir, or manifest should be enabled
-    # See http://docs.puppetlabs.com/puppet/latest/reference/environments.html#enabling-directory-environments
-    #   for more information on the changes
-        environmentpath = $confdir/environments
+The evolution to using Environments in this way was a gradual one from community
```
Comment (mwaggett) [inline doc/dynamic-environments/master-configuration.mkd]: nitpick, but you capitalize "Environments" here, but not above. imo, we should be consistent about whether we capitalize that.

**Review by mwaggett — COMMENTED** [review]
this generally looks good! just a couple nits 😅

**Comment by baurmatt** [comment]
This looks way better than before, thank you! ❤️

**Comment by baurmatt** [comment]
I've just found another to links which aren't valid anymore. Perhaps you could also fix them with this PR? :)

https://github.com/puppetlabs/r10k/blob/f87ec83f58ad04c8b5c4a0a59cc9af660c48da42/lib/r10k/cli/deploy.rb#L21

https://github.com/puppetlabs/r10k/blob/5389b9ff51d0a271a9cdbd4442c134c9149e12aa/README.mkd#L11

**Comment by justinstoller** [comment]
I couldn't find that blog post, nor Adrien's website where he first outlined it. I've instead linked to the Puppet's "About Environments" page which is probably the canonical source now.

File: doc/dynamic-environments/master-configuration.mkd (hunk 2 above)
Comment (binford2k) [reply doc/dynamic-environments/master-configuration.mkd]: I vote lowercase

Comment (justinstoller) [reply doc/dynamic-environments/master-configuration.mkd]: fixed

**Review by mwaggett — DISMISSED** [review]

**Comment by mwaggett** [comment]
Wanna squash @justinstoller ?

**Comment by justinstoller** [comment]
squashed

**Review by mwaggett — APPROVED** [review]

**Pull request merged**